app/src/utils/utils.tsx

```diff
@@ -0,0 +1,47 @@
+import {useNavigate} from "react-router-dom";
+import './utils.css';
+
+interface SearchBarProps {
+    isHeader: boolean;
+}
+
+// Will change depending if the search bar is for the homepage or not
+export const SearchBar: React.FC<SearchBarProps> = ({ isHeader }) => {
+    const navigate = useNavigate();
+
+    const performSearch = () => {
+        const searchInput = document.querySelector<HTMLInputElement>(
+            isHeader ? '.header-search-bar' : '.search-bar'
+        );
+        if (searchInput && searchInput.value !== '') {
+            const searchString = searchInput.value;
+
+            // Replaces all strings and instances of CSE
+            const stringWithoutSpacesOrCSE = searchString.replace(/\s|CSE|cse/g, '');
+
+            navigate(`/search/${stringWithoutSpacesOrCSE}`);
+        }
+    };
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === 'Enter') {
+            performSearch();
+        }
+    };
+
+    const handleSearch = () => {
+        performSearch();
+    };
+
+    return (
+        <div className={isHeader ? "header-search-container" : "search-container"}>
+            <input
+                type="text"
+                placeholder={isHeader ? "Search..." : "Look up a CSE class"}
```

Comment:
NIT: A user might assume that a search bar could look up a class by _name_. Should we specify here or somewhere else 'look up a CSE class by number'?